### PR TITLE
Add payment index to payviapaymentrequest and subscribetopayviarequest

### PR DESCRIPTION
**Pull request by niteshbalusu11**

### CODE REVIEW

**Comment by niteshbalusu11** [comment]
Fixed, comments, typescript and test cases.

File: lnd_responses/pending_from_payment.js [hunk 1]
```diff
@@ -67,6 +67,7 @@ const nsAsDate = ns => new Date(Number(BigInt(ns) / BigInt(1e6)));
     created_at: <Payment Created At ISO 8601 Date String>
     destination: <Payment Destination Public Key Hex String>
     id: <Payment Hash Hex String>
+    index: <Payment Index String>
```
Comment (alexbosworth) [inline lnd_responses/pending_from_payment.js:70]:
```suggestion
    index: <Payment Index Offset Number String>
```

File: lnd_responses/confirmed_from_payment.js [hunk 2]
```diff
@@ -81,6 +81,7 @@ const nsAsDate = ns => new Date(Number(BigInt(ns) / BigInt(1e6)));
       timeout: <First Path Timeout Block Height Number>
     }]
     id: <Payment Hash Hex String>
+    index: <Payment Index String>
```
Comment (alexbosworth) [inline lnd_responses/confirmed_from_payment.js:84]:
```suggestion
    index: <Payment Index Offset Number String>
```

File: lnd_methods/offchain/finished_payment.js [hunk 3]
```diff
@@ -80,6 +80,7 @@ const {returnResult} = require('asyncjs-util');
       timeout: <Timeout Block Height Number>
     }]
     [id]: <Payment Hash Hex String>
+    index: <Payment Index String>
```
Comment (alexbosworth) [inline lnd_methods/offchain/finished_payment.js:83]:
```suggestion
    index: <Payment Index Offset Number String>
```

**Comment by alexbosworth** [comment]
I added a comment on the attribute doc description, generally in those i try to kind of restate the field but in a different way than the actual naming

So I think this should be lining up with the offset number you'd see in the list of total payments

**Pull request merged**